runtime/near-wallet-contract/implementation/wallet-contract/src/tests/utils/test_context.rs

```diff
@@ -0,0 +1,243 @@
+                "target": target,
+                "tx_bytes_b64": codec::encode_b64(&codec::rlp_encode(tx))
+            }))
+            .max_gas()
+            .transact()
+            .await?
+            .into_result()?
+            .json()?;
+
+        Ok(result)
+    }
+
+    pub async fn get_nonce(&self) -> anyhow::Result<u64> {
+        let nonce: U64 = self.inner.view(GET_NONCE).await?.json()?;
+        Ok(nonce.0)
+    }
+
+    /// Add a new `FunctionCall` access key to the Wallet Contract.
+    /// The idea is that this allows the relayer to submit transactions signed by
+    /// the Wallet Contract directly.
+    pub async fn register_relayer(
+        &mut self,
+        worker: &Worker<Sandbox>,
+    ) -> anyhow::Result<PublicKey> {
+        let relayer_account = worker.dev_create_account().await?;
+        let relayer_key = SecretKey::from_random(KeyType::ED25519);
+        let relayer_pk = relayer_key.public_key();
+
+        let action = Action::AddKey {
+            public_key_kind: 0,
+            public_key: relayer_pk.key_data().to_vec(),
+            nonce: 0,
+            is_full_access: false,
+            is_limited_allowance: false,
+            allowance: 0,
+            receiver_id: self.inner.id().to_string(),
+            method_names: vec![RLP_EXECUTE.into()],
+        };
+        let nonce = self.get_nonce().await?;
+        let signed_transaction =
+            utils::create_signed_transaction(nonce, self.inner.id(), Wei::zero(), action, &self.sk);
+
+        // Call the Wallet Contract from the relayer account to add the key
+        let result: ExecuteResponse = relayer_account
+            .call(self.inner.id(), RLP_EXECUTE)
+            .args_json(serde_json::json!({
+                "target": self.inner.id(),
+                "tx_bytes_b64": codec::encode_b64(&codec::rlp_encode(&signed_transaction))
+            }))
+            .max_gas()
+            .transact()
+            .await?
+            .into_result()?
+            .json()?;
+
+        assert!(result.success, "Adding Relayer's key failed: {:?}", result.error);
+
+        // Tell near-workspaces to use this new key instead when
+        // signing transactions from the Wallet Contract
+        self.inner.as_account_mut().set_secret_key(relayer_key);
+
+        Ok(relayer_pk)
+    }
+}
+
+pub struct TestContext {
+    pub worker: Worker<Sandbox>,
+    pub wallet_contract: WalletContract,
+    pub wallet_sk: near_crypto::SecretKey,
+    pub wallet_address: Address,
+    pub address_registrar: Contract,
+    pub wallet_contract_bytes: Vec<u8>,
+}
+
+impl TestContext {
+    pub async fn new() -> anyhow::Result<Self> {
+        let _guard = LOCK.lock().await;
+        let worker = near_workspaces::sandbox().await?;
+
+        let address_registrar = Self::deploy_address_registrar(&worker).await?;
+        let wallet_contract_bytes = build_contract(BASE_DIR, PACKAGE_NAME).await?;
+        // Restore address registrar account id file
```

Comment:
Why do we need restoring here? `git checkout` looks fragile here and may cause issues if later used in CI where git commands are restricted.

Reply:
In the `near-workspaces` test the registrar contract is deployed to a random account (using  `dev-deploy`), so it needs to be dynamically set when compiling the wallet contract. This is handled by reading the account id from a file at compile time. Therefore that file is modified during the tests, but it should not stay modified after the tests are finished. An alternative to `git checkout` could be to read the file before changing it and remember the old value to restore manually. I can make this change.

Reply:
Done in https://github.com/near/nearcore/pull/10850/commits/b900f78f03b35c7aa330871fe671090a30353b4c